public/app/features/plugins/extensions/utils.tsx

```diff
@@ -446,6 +446,33 @@ export const getExtensionPointPluginDependencies = (extensionPointId: string): s
     }, []);
 };
 
+export const getExtensionPointPluginMeta = (extensionPointId: string) => {
+  return new Map(
+    getExtensionPointPluginDependencies(extensionPointId)
+      .map((pluginId) => {
+        const app = config.apps[pluginId];
+        // if the plugin does not exist or does not expose any components or links to the extension point, return undefined
+        if (
+          !app ||
+          (!app.extensions.addedComponents.some((component) => component.targets.includes(extensionPointId)) &&
+            !app.extensions.addedLinks.some((link) => link.targets.includes(extensionPointId)))
+        ) {
+          return undefined;
+        }
+        return [
+          pluginId,
+          {
+            exposedComponents: app.extensions.addedComponents.filter((component) =>
+              component.targets.includes(extensionPointId)
+            ),
+            addedLinks: app.extensions.addedLinks.filter((link) => link.targets.includes(extensionPointId)),
+          },
+        ] as const;
+      })
+      .filter((c): c is NonNullable<typeof c> => c !== undefined)
+  );
+};
```

Comment:
Would be nice to add some tests for this utility function. 🙏

Reply:
Definitely. Just wanted to get alignment first and see if the approach is viable - will add tests today 👍

Reply:
Added tests in 1edcaca5df3bda120394b8befa21592f69a422af